### PR TITLE
 Fix: HierarchyCircuitBreakerTelemetryTests testCircuitBreakerTripCountMetric failing

### LINKED ISSUE #111778: [CI] HierarchyCircuitBreakerTelemetryTests testCircuitBreakerTripCountMetric failing
**Build Scans:**
- [elasticsearch-intake #8873 / part1](https://gradle-enterprise.elastic.co/s/g7t3b2jzib53i)
- [elasticsearch-periodic-platform-support #3725 / almalinux-8-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/4x5i5wybi6dsi)
- [elasticsearch-periodic-platform-support #3725 / ubuntu-2004-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/hzwi64jwdjxmi)
- [elasticsearch-periodic #3749 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/s3joj5jkcdhuk)
- [elasticsearch-periodic-platform-support #3719 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/5set6pevkcht6)
- [elasticsearch-periodic-platform-support #3716 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/bbdgua5khv5n6)
- [elasticsearch-periodic-platform-support #3713 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/nrchxgjbg72w6)
- [elasticsearch-periodic-platform-support #3707 / sles-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/d5o62pu3647yg)
- [elasticsearch-periodic-platform-support #3707 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/4jgmlx452l6j2)

**Reproduction Line:**
```
./gradlew ":server:test" --tests "org.elasticsearch.indices.breaker.HierarchyCircuitBreakerTelemetryTests.testCircuitBreakerTripCountMetric" -Dtests.seed=BD2A56CFEFCFAA6A -Dtests.locale=sr-RS -Dtests.timezone=PLT -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testCircuitBreakerTripCountMetric'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.indices.breaker.HierarchyCircuitBreakerTelemetryTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: safeGet: listener was completed exceptionally
```

**Issue Reasons:**
- [main] 9 failures in test testCircuitBreakerTripCountMetric (3.7% fail rate in 242 executions)
- [main] 5 failures in pipeline elasticsearch-periodic-platform-support (71.4% fail rate in 7 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

### PR DESCRIPTION
**Pull request by ldematte**
This test started failing consistently after https://github.com/elastic/elasticsearch/pull/111744 was merged; the error is raised during test teardown, in the common cluster cleanup code.
The message/line number for failure indicates that a ExecutionException is thrown. Root cause is a CircuitBreakingException; it looks like the change exposed a hidden problem in the test, as the test was not resetting the cluster settings in cleanup.

I added a reset method to cleanup settings; I also cleaned up the test a little, removing some dead code and one unneeded class, and moved it to a more appropriate location/name (it is a IT test). Probably some more cleanup could be done, but I wanted to keep the changes down so I did the minimum to put it "in a good shape".

Closes https://github.com/elastic/elasticsearch/issues/111778

### CODE REVIEW

**Review by rjernst — COMMENTED** [review]
There seems to be several things going on in this change. Can you comment on which parts address the test bug, and which parts are cleanup?

File: server/src/internalClusterTest/java/org/elasticsearch/indices/memory/breaker/HierarchyCircuitBreakerTelemetryIT.java [hunk 1]
```diff
@@ -41,54 +39,11 @@
...
 @ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0, numClientNodes = 0, supportsDedicatedMasters = true)
-public class HierarchyCircuitBreakerTelemetryTests extends ESIntegTestCase {
+public class HierarchyCircuitBreakerTelemetryIT extends ESIntegTestCase {
 
     @Override
     protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return List.of(TestCircuitBreakerTelemetryPlugin.class);
-    }
-
-    public static class TestCircuitBreakerTelemetryPlugin extends TestTelemetryPlugin {
```
Comment (ldematte) [inline server/src/internalClusterTest/java/org/elasticsearch/indices/memory/breaker/HierarchyCircuitBreakerTelemetryIT.java:51]: Cleanup: we don't need to subclass `TestTelemetryPlugin`, it's possible (and better) to use it directly

File: server/src/internalClusterTest/java/org/elasticsearch/indices/memory/breaker/HierarchyCircuitBreakerTelemetryIT.java [hunk 2]
```diff
@@ -142,37 +97,29 @@ public void testCircuitBreakerTripCountMetric() {
         fail("Expected exception not thrown");
     }
 
-    private List<Measurement> getMeasurements(String dataNodeName) {
-        final TestTelemetryPlugin dataNodeTelemetryPlugin = internalCluster().getInstance(PluginsService.class, dataNodeName)
-            .filterPlugins(TestCircuitBreakerTelemetryPlugin.class)
+    @After
+    public void resetClusterSetting() {
```
Comment (ldematte) [inline server/src/internalClusterTest/java/org/elasticsearch/indices/memory/breaker/HierarchyCircuitBreakerTelemetryIT.java:101]: Fix: we need to reset cluster settings after the tests, before the global IT tests cleanup; having a very strict circuit breaker during global IT tests cleanup will trip it and break the tests

File: server/src/internalClusterTest/java/org/elasticsearch/indices/memory/breaker/HierarchyCircuitBreakerTelemetryIT.java [hunk 3]
```diff
@@ -142,37 +97,29 @@ public void testCircuitBreakerTripCountMetric() {
...
         return Measurement.combine(
-            Stream.of(dataNodeTelemetryPlugin.getLongCounterMeasurement(CircuitBreakerMetrics.ES_BREAKER_TRIP_COUNT_TOTAL).stream())
+            Stream.of(telemetryPlugin.getLongCounterMeasurement(CircuitBreakerMetrics.ES_BREAKER_TRIP_COUNT_TOTAL).stream())
                 .flatMap(Function.identity())
                 .toList()
         );
     }
-
-    // Make sure circuit breaker telemetry on trip count reports the same values as circuit breaker stats
-    private void assertCircuitBreakerTripCount(
```
Comment (ldematte) [inline server/src/internalClusterTest/java/org/elasticsearch/indices/memory/breaker/HierarchyCircuitBreakerTelemetryIT.java:158]: This method was unused, likely a leftover from before adopting `TestTelemetryPlugin`

**Comment by ldematte** [comment]
I added some comments to explain what is what and why

**Review by rjernst — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**